File: VB/WindowsApplication1/MyMemoEditScrollBarViewInfo.vb
Imports Microsoft.VisualBasic
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.ViewInfo

Namespace WindowsApplication1
	Public Class MyMemoEditScrollBarViewInfo
		Inherits ScrollBarViewInfo
		Private fWidth As Integer

		Public Sub New(ByVal SB As IScrollBar)
			MyBase.New(SB)
			If ScrollBarType = ScrollBarType.Horizontal Then
				fWidth = SystemInformation.HorizontalScrollBarArrowWidth
			Else
				fWidth = SystemInformation.VerticalScrollBarArrowHeight
			End If
		End Sub

		Public Overrides ReadOnly Property ButtonWidth() As Integer
			Get
				If fWidth > ScrollBarWidth * 2 Then
					Return 0
				End If
				If fWidth > ScrollBarWidth / 2 Then
					fWidth = ScrollBarWidth / 2
				End If
				Return fWidth
			End Get
		End Property
		Public Sub SetButtonWidth(ByVal Width As Integer)
			If Width >= 0 Then
				fWidth = Width
				CalculateCore()
			End If
		End Sub
	End Class

End Namespace

File: CS/WindowsApplication1/MyMemoEditScrollBarViewInfo.cs
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.ViewInfo;

namespace WindowsApplication1
{
    public class MyMemoEditScrollBarViewInfo : ScrollBarViewInfo
    {
        int fWidth;

        public MyMemoEditScrollBarViewInfo(IScrollBar SB)
            : base(SB)
        {
            if (ScrollBarType == ScrollBarType.Horizontal)
                fWidth = SystemInformation.HorizontalScrollBarArrowWidth;
            else
                fWidth = SystemInformation.VerticalScrollBarArrowHeight;
        }

        public override int ButtonWidth
        {
            get
            {
                if (fWidth > ScrollBarWidth * 2)
                    return 0;
                if (fWidth > ScrollBarWidth / 2)
                    fWidth = ScrollBarWidth / 2;
                return fWidth;
            }
        }
        public void SetButtonWidth(int Width)
        {
            if (Width >= 0)
            {
                fWidth = Width;
                CalculateCore();
            }
        }
    }

}
